### ISSUE TITLE
Critical, fatal error 'NoneType' object has to no attribute 'read'

### ISSUE THREAD

**Issue by PhistterbutInc** [mariolukas/FabScanPi-Server#128]
## *Who* is the bug affecting?

User

## *What* is affected by this bug?

The server does not start

## *When* does this occur?

every time I reboot and try to start or restart the fabscanpi-server service

## *Where* on the platform does it happen?

I'm using the latest image as of 4/26/19 and a Rpi zero W. I forgot to mention I'm using an arduino mega and usb cable connection.

## *How* do we replicate the issue?

Burn disk using win32 disk imager, insert SD into slot and power on the pi zero w. I've also tried to install the software manually as listed under the getting started page by adding the key and repo.

## Expected behavior (i.e. solution)

It should have worked considering I'm using the stock image provided. I didn't expect errors.

## Other Comments
Restarting the server service doesn't work either. Screen cap of errors included.
![error_log_fabscanpi](https://user-images.githubusercontent.com/30960886/56844126-9fe54100-6868-11e9-8191-683ff4cbcdd9.jpg)

I'm using the latest version of chrome by the way.
Here's a photo of the webpage. I've done what it says and it prints the message shown on the webpage just as it's written.
![error_webpage](https://user-images.githubusercontent.com/30960886/56844167-3dd90b80-6869-11e9-9bd9-bf2534cbccf2.jpg)

UPDATE:
~~~~~~~
Further info. I've been able to use the command: 

sudo avrdude-autoreset -p m2560 -b 115200 -c wiring -P/dev/ttyACM0

 to get the following output:

avrdude-original: Using autoreset DTR on GPIO Pin 7

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.01s

avrdude: Device signature = 0x1e9801 (probably m2560)

avrdude: safemode: Fuses OK (E:FD, H:D8, L:FF)

avrdude done.  Thank you.

Is there anyway to use 'wiring' to communicate using the server front end instead of avrdude-original???
I changed the baud rate in the firmware and config file to 115200 for both connection and firmware flashing bauds. The web page still shows no GUI for me, just the error message.

**Comment by mariolukas** [comment]
Have you checked the serial port config in your default.config.json. The default value is

`/dev/ttyAMA0`

but in your case it should be 

`/dev/ttyACM0`

**Comment by PhistterbutInc** [comment]
Yes, I've changed that value to /dev/ttyACM0

I've also changed the command in /etc/init.d/fabscanpi-server, the line at avrdude-autoreset, to avrdude-autoreset -c wiring -p m2560 -P /dev/ttyACM0 -b 115200 -l /dev/null

and still a no go. I also changed the baud rate in the firmware and default.config.json to 115200.

During all of my testing the RAMPS board has been plugged into the pi zero W.

I also changed the file under /usr/bin/avrdude-reset to ACM0 instead of AMA0
![avrdude-reset](https://user-images.githubusercontent.com/30960886/57077462-58681600-6ca9-11e9-80e8-eaca6cf17c64.png)

Not sure if it's related or not but I've also tried installing manually on top of the pre-made image and when I try installing 'python-opencv-tbb' it gives an error as seen in the screen cap.
![broken_packages](https://user-images.githubusercontent.com/30960886/57081544-2d35f480-6cb2-11e9-9604-0de7d3f09eff.png)
